mlx/backend/metal/quantized.cpp

```diff
@@ -41,8 +41,35 @@ void QuantizedMatmul::eval_gpu(const std::vector<array>& inputs, array& out) {
   int B = x.size() / D;
   int O = out.shape(-1);
```

Comment:
Do we want to add a check here (or in the op) to make sure that the input / output are `float` ? We can land up with an unclear error message otherwise